### PR TITLE
Add a success dialog to window renaming

### LINKED ISSUE #9804: Renaming the window should display confirmation
### Windows Terminal version (or Windows build number)

1.8.1022.0

### Other Software

_No response_

### Steps to reproduce

Rename the window using the `renameWindow` keybinding:

```
{ "command": { "action": "renameWindow", "name": "Foo" }, "keys": "ctrl+u" },
```

### Expected Behavior

Some kind of confirmation that my window has been renamed.

### Actual Behavior

Nothing appears on the screen.

### PR DESCRIPTION
**Pull request by zadjii-msft**
I added a `RenameSucceededText` property to the `TerminalPage` which returns the
formatted message `Successfully renamed window to "{WindowNameForDisplay()}"`

This _doesn't_ pop the dialog when you `wt -w foo` for the first time. Only
_subsequent_ renames.

## References
* Added in #9662
* Closes #9804

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -2601,15 +2601,65 @@ namespace winrt::TerminalApp::implementation
...
+        // display the success toast.
+        co_await resume_foreground(Dispatcher());
+        if (auto page{ weakThis.get() })
+        {
+            if (changed)
+            {
+                page->_PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"WindowNameForDisplay" });
+                page->_PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"RenameSucceededText" });
+
+                // DON'T display the confirmation if this is the name we were
```
Comment (miniksa) [inline src/cascadia/TerminalApp/TerminalPage.cpp:2623]: hah that's a hilarious bug

**Review by miniksa — APPROVED** [review]

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw [hunk 2]
```diff
@@ -618,10 +618,17 @@
   <data name="RenameFailedToast.Subtitle" xml:space="preserve">
     <value>Another window with that name already exists</value>
   </data>
+  <data name="RenameSucceededToast.Title" xml:space="preserve">
+    <value>Successfully renamed window</value>
+  </data>
+  <data name="RenameSucceededText" xml:space="preserve">
+    <value>Successfully renamed window to "{}"</value>
+    <comment>{} will be replaced by the new name assigned by the user</comment>
+  </data>
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/Resources/en-US/Resources.resw:627]: This seems a bit redundant. @cinnamon-msft mind taking a look?

**Review by carlos-zamora — COMMENTED** [review]
I'm fine with this. I just don't like the content of the confirmation toast; feels redundant, but idk how to fix it. Once that gets figured out, I'll approve.

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw (hunk 2 above)
Comment (cinnamon-msft) [reply src/cascadia/TerminalApp/Resources/en-US/Resources.resw:627]: Yeah I was just thinking we could pop up the same toast as the `identifyWindow` command. It does the same thing without having to localize more stuff. :)

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Resources/en-US/Resources.resw]: this is a _great_ idea

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw [hunk 3]
```diff
@@ -1,17 +1,17 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/Resources/en-US/Resources.resw:1]: nit: undo changes to this file

File: src/cascadia/TerminalApp/TerminalPage.h [hunk 4]
```diff
@@ -327,6 +327,7 @@ namespace winrt::TerminalApp::implementation
         void _WindowRenamerActionClick(const IInspectable& sender, const IInspectable& eventArgs);
         void _RequestWindowRename(const winrt::hstring& newName);
         void _WindowRenamerKeyUp(const IInspectable& sender, winrt::Windows::UI::Xaml::Input::KeyRoutedEventArgs const& e);
+        void _DisplayRenameSuccessToast();
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.h:330]: This doesn't exist anymore?

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 5]
```diff
@@ -2606,12 +2606,33 @@ namespace winrt::TerminalApp::implementation
...
+        // On the foreground thread, raise property changed notifications, and
+        // display the success toast.
+        co_await resume_foreground(Dispatcher());
+        if (auto page{ weakThis.get() })
+        {
+            if (changed)
+            {
+                page->_PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"WindowName" });
+                page->_PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"WindowNameForDisplay" });
+                page->_PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"RenameSucceededText" });
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp:2627]: This one doesn't exist anymore?

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Small things. If possible, update the gif in the OP too please.

Also, are we gonna service this?

**Comment by DHowett** [comment]
moved from body window-rename-success-000

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
@carlos-zamora I addressed your nits

**Review by carlos-zamora — APPROVED** [review]

**Pull request merged**